cpp/tests/sorted-merge/sorted_merge-test.cu

```diff
@@ -0,0 +1,199 @@
+                                                cudf::table(rightColumns, 2),
+                                                sortByCols,
+                                                ordersDeviceVector);
+
+    const gdf_size_type outputLength = leftColumn1->size + rightColumn1->size;
+    cudf::test::column_wrapper<TypeParam> expectedDataWrap1({0, 1, 1, 2, 2, 3}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(outputLength),0xFF));
+    cudf::test::column_wrapper<TypeParam> expectedDataWrap2({4, 5, 8, 6, 9, 7}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(outputLength),0xFF));
+
+    EXPECT_TRUE(gdf_equal_columns<TypeParam>(expectedDataWrap1.get(), outputTable.get_column(0)));
+    EXPECT_TRUE(gdf_equal_columns<TypeParam>(expectedDataWrap2.get(), outputTable.get_column(1)));
+}
+
+TYPED_TEST(SortedMergeTest, Merge2KeyColumns) {
+    cudf::test::column_wrapper<TypeParam> leftColWrap1({0, 1, 2, 3}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(4),0xFF));
+    cudf::test::column_wrapper<TypeParam> leftColWrap2({4, 5, 6, 7}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(4),0xFF));
+    gdf_column *leftColumn1 = leftColWrap1.get();
+    gdf_column *leftColumn2 = leftColWrap2.get();
+
+    cudf::test::column_wrapper<TypeParam> rightColWrap1({1, 2}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(2),0xFF));
+    cudf::test::column_wrapper<TypeParam> rightColWrap2({8, 9}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(2),0xFF));
+    gdf_column *rightColumn1 = rightColWrap1.get();
+    gdf_column *rightColumn2 = rightColWrap2.get();
+
+    gdf_column *leftColumns[]  = {leftColumn1, leftColumn2};
+    gdf_column *rightColumns[] = {rightColumn1, rightColumn2};
+
+    std::vector<gdf_size_type> sortByCols = {0, 1};
+    rmm::device_vector<int8_t> ordersDeviceVector(std::vector<int8_t>{GDF_ORDER_ASC, GDF_ORDER_DESC});
+
+    cudf::table outputTable = cudf::sorted_merge(cudf::table(leftColumns, 2),
+                                                cudf::table(rightColumns, 2),
+                                                sortByCols,
+                                                ordersDeviceVector);
+
+    const gdf_size_type outputLength = leftColumn1->size + rightColumn1->size;
+    cudf::test::column_wrapper<TypeParam> expectedDataWrap1({0, 1, 1, 2, 2, 3}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(outputLength),0xFF));
+    cudf::test::column_wrapper<TypeParam> expectedDataWrap2({4, 8, 5, 9, 6, 7}, std::vector<gdf_valid_type>(gdf_valid_allocation_size(outputLength),0xFF));
+
+    EXPECT_TRUE(gdf_equal_columns<TypeParam>(expectedDataWrap1.get(), outputTable.get_column(0)));
+    EXPECT_TRUE(gdf_equal_columns<TypeParam>(expectedDataWrap2.get(), outputTable.get_column(1)));
+}
+
+TYPED_TEST(SortedMergeTest, Merge2KeyNullColumns) {
+    cudf::test::column_wrapper<TypeParam> leftColWrap1({0, 1, 2, 3, -1}, bools_to_valids({1, 1, 1, 1, 0}));
```

Comment:
Why not use a `column_wrapper` constructor that takes generator lambdas -- then you won't have to have `bools_to_valids`, it does it for you.